### ISSUE TITLE
Matching a mix of dynamic and static routes with overlaps

### ISSUE THREAD

**Issue by loony-bean** [ibraheemdev/matchit#12]
Hi @ibraheemdev! First of all, thanks for your work!

I'm trying to match a mix of dynamic and static routes, which have overlapping parts. But not all the cases are matching.

I guess this is better explained with the following code snippet.

```
#[cfg(test)]
mod tests {
    use matchit::Node;

    #[test]
    fn test_routes() {
        let mut matcher = Node::new();

        matcher.insert("/:object/:id", "object with id").unwrap();
        matcher.insert("/secret/:id/path", "secret with id and path").unwrap();

        let matched = matcher.at("/secret/978/path").unwrap();
        assert_eq!(matched.params.get("id"), Some("978"));

        let matched = matcher.at("/something/978").unwrap();
        assert_eq!(matched.params.get("id"), Some("978"));
        assert_eq!(matched.params.get("object"), Some("something"));

        let _ = matcher.at("/secret/978").unwrap(); // panics
    }
}
```

It it expected behaviour?

**Comment by ibraheemdev** [comment]
Hi, that is not expected behavior. It looks like the node never backtracks in that case.. I'll look into it.

**Comment by ibraheemdev** [comment]
Fixed in (329e24e16592d2dc7eb9ee9066a2f6d7e8f7b908). I'll add some more tests and get a patch release out.
